### DISCUSSION TITLE
What kind of color is this?

### DISCUSSION THREAD

**Discussion by RYoder97** [jasonacox/tinytuya#548] [Q&A] [answered]
My bulb that uses DPS 24 (colour_data_v2) returns a string like this: 00ac03b103ce

What on earth is this type of color? To my understanding, the bulbs use HSV. devices.json mapping would give the impression you can simply pass a dict like this:

```json
{
   "h":80,
   "s":80,
   "v":80
}
```

But passing an object like that does nothing.

Looking to see how to set color using the method ```device.set_value``` by the way.

**Comment by jasonacox** [comment]
Check out the SmartBulb example: https://github.com/jasonacox/tinytuya/blob/master/examples/bulb.py and then the source on setting the DPS for hsv: https://github.com/jasonacox/tinytuya/blob/11f43ccbd23c507a8a0fbcd3e96ead447f4d8ee0/tinytuya/BulbDevice.py#L351-L392

**Reply by RYoder97** [reply]
> Check out the SmartBulb example: https://github.com/jasonacox/tinytuya/blob/master/examples/bulb.py and then the source on setting the DPS for hsv:
> 
> https://github.com/jasonacox/tinytuya/blob/11f43ccbd23c507a8a0fbcd3e96ead447f4d8ee0/tinytuya/BulbDevice.py#L351-L392

My bad, I was a bit confused for a while because I was looking for a colorsys.py file in the source code, but I realize now that it's a built-in library.

**Comment by uzlonewolf** [comment, marked as answer]
> 00ac03b103ce

0x00ac = 172 (H, 0-360)
0x03b1 = 945 = 94.5% (S, 0.0-100.0%)
0x03ce = 974 = 97.4% (V, 0.0-100.0%)

Even if you don't instantiate a BulbDevice object you can still use the class functions:
```python
import tinytuya

hexvalue = '00ac03b103ce'

print( tinytuya.BulbDevice._hexvalue_to_hsv(hexvalue, bulb='B') )
# (0.4777777777777778, 0.945, 0.974)

print( tinytuya.BulbDevice._hexvalue_to_rgb(hexvalue, bulb='B') )
# (13, 248, 217)

print( tinytuya.BulbDevice._rgb_to_hexvalue(255, 128, 64, bulb='B') )
# 001402ed03e8
```

@jasonacox What do you think of promoting those 3 functions to officially supported?

**Reply by jasonacox** [reply]
That's a great idea @uzlonewolf - I've caught myself doing what you just did multiple times.

**Reply by RYoder97** [reply]
Interesting. In hindsight, I should have probably just used BulbDevice and all the other pre-coded classes. I'll probably end up recoding all the Python at some point and store the device category in my Python global ```devices``` variable.
